integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -121,9 +121,12 @@ object DataLoadProcessBuilderOnSpark {
         CarbonProperties.getInstance().getGlobalSortRddStorageLevel()))
     }
 
+    val sortColumnIndex = configuration.getSortColumnRangeInfo.getSortColumnIndex
```

Comment:
don't use configuration.getSortColumnRangeInfo.getSortColumnIndex, when sort_scope is global sort, configuration.getSortColumnRangeInfo is null. 
Use:
`val numberOfSortColumns = configuration.getNumberOfSortColumns`,
and change CarbonRow.getKey(int[] sortColumnIndex) to getKey(int numberOfSortColumn).